Engine/DataFeeds/UniverseSelection.cs

```diff
@@ -103,6 +104,51 @@ public SecurityChanges ApplyUniverseSelection(Universe universe, DateTime dateTi
 
                     // perform the fine fundamental universe selection
                     selectSymbolsResult = fineFiltered.FineFundamentalUniverse.PerformSelection(dateTimeUtc, fineCollection);
+
+                    // WARNING -- HACK ATTACK -- WARNING
+                    // Fine universes are considered special due to their chaining behavior.
+                    // As such, we need a means of piping the fine data read in here back to the data feed
+                    // so that it can be properly emitted via a TimeSlice.Create call. There isn't a mechanism
+                    // in place for this function to return such data. The following lines are tightly coupled
+                    // to the universeData dictionaries in SubscriptionSynchronizer and LiveTradingDataFeed and
+                    // rely on reference semantics to work.
+
+                    var coarseData = universeData.Data.OfType<CoarseFundamental>().ToDictionary(d => d.Symbol);
+                    universeData.Data = new List<BaseData>();
+                    foreach (var fine in fineCollection.Data.OfType<FineFundamental>())
```

Comment:
Tried pretty hard to consolidate how we manage fine to be just like every other universe to no avail. This code is [smelly](https://en.wikipedia.org/wiki/Code_smell) but works. Despite the stench factor, it actually made the merge operation below a breeze (otherwise it would have probably been implemented in the `Security.Fundamentals` property getter by combining the coarse/fine at invocation time.